ecosystem/platform/server/app/components/claim_nft_button_component.html.erb

```diff
@@ -2,11 +2,22 @@
   <p class="text-red-400 hidden mb-8" data-claim-nft-target="transactionFailedError">
   NFT minting failed. Please ensure that your account has sufficient tokens to perform the transaction and try again.
   </p>
-  <%= render ButtonComponent.new(href: nft_offer_path(@nft_offer, wallet: @wallet.public_key, wallet_name: @wallet.wallet_name), data: { action: 'claim-nft#handleClick' }, **@rest) do %>
-    <% if content %>
-      <%= content %>
+  <%= form_with url: nft_offer_path(@nft_offer), method: 'PUT', data: { controller: 'recaptcha', action: 'recaptcha#validate claim-nft#handleSubmit', claim_nft_target: 'form' } do |form| %>
+    <%= form.hidden_field :wallet, value: @wallet.public_key %>
+    <%= form.hidden_field :wallet_name, value: @wallet.wallet_name %>
+    <%= render ButtonComponent.new(**@rest) do %>
+      <% if content %>
+        <%= content %>
+      <% else %>
+        Claim NFT
+      <% end %>
+    <% end %>
+    <% if @recaptcha_v2 %>
```

Comment:
curios, what is the benefit to use v2 over v3?

Reply:
It's only there as a fallback if v3 doesn't succeed.